2a_model/src/models/Snakefile_base.smk

```diff
@@ -29,7 +30,7 @@ rule as_run_config:
 
 rule prep_io_data:
     input:
-        "../../../out/well_obs_io.zarr",
+        "../../../out/"+site_set+"_io.zarr",
```

Comment:
Maybe personal preference, but I think this is a little be cleaner. Using Python's string formatting. That way you don't have to open and close the quotes and I think it's a little less error prone since it's harder to accidentally add in an extra space.
```suggestion
        f"../../../out/{site_set}_io.zarr",
```